shotgun_api3/shotgun.py

```diff
@@ -29,36 +29,63 @@
  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 """
 
+# ---- imports common to py2 & py3
+
 import base64
-import cookielib    # used for attachment upload
-import cStringIO    # used for attachment upload
+import copy
 import datetime
+import json
 import logging
-import mimetools    # used for attachment upload
 import os
 import re
-import copy
+import shutil       # used for attachment download
 import stat         # used for attachment upload
 import sys
 import time
-import types
-import urllib
-import urllib2      # used for image upload
-import urlparse
-import shutil       # used for attachment download
-import httplib      # Used for secure file upload.
-
-# use relative import for versions >=2.5 and package import for python versions <2.5
-if (sys.version_info[0] > 2) or (sys.version_info[0] == 2 and sys.version_info[1] >= 6):
-    from sg_26 import *
-elif (sys.version_info[0] > 2) or (sys.version_info[0] == 2 and sys.version_info[1] >= 5):
-    from sg_25 import *
-else:
-    from sg_24 import *
 
-# mimetypes imported in version specific imports
-mimetypes.add_type('video/webm','.webm') # webm and mp4 seem to be missing
-mimetypes.add_type('video/mp4', '.mp4')  # from some OS/distros
+# ---- py2/py3 compatibility
+
+import six
+
+# ---- imports that differ between 2 & 3
+
+from six.moves import http_cookiejar as cookielib
+from six.moves.urllib import request as url_request
+from six.moves.urllib.error import HTTPError, URLError
+from six.moves.urllib.parse import (
+    quote,
+    splituser,
+    unquote,
+    urljoin,
+    urlsplit,
+    urlunparse,
+)
+from six.moves import xmlrpc_client
+
+# import errors used by this api
+from .errors import (
+    ShotgunError,
+    ShotgunFileDownloadError,
+    Fault,
+    AuthenticationFault,
+    MissingTwoFactorAuthenticationFault,
+    UserCredentialsNotAllowedForOxygenAuthenticationFault,
+    UserCredentialsNotAllowedForSSOAuthenticationFault
+)
+
+# form post handler utility
+from .form_post_handler import FormPostHandler
+
+# client/server capability apis
+from .client_caps import ClientCapabilities
+from .server_caps import ServerCapabilities
+
+# lib imports
+from .lib.httplib2 import Http, ProxyInfo, socks, ssl_error_classes
```

Comment:
recommend close review of `ssl_error_classes` here. this was an attempt to account for `SSLHandshakeError` not existing in py3 version of `httplib2`.